scripts/examples/03-Machine-Learning/00-TensorFlow/tf_object_detection.py

```diff
@@ -36,30 +36,49 @@
     (255, 255, 255),
 ]
 
+# FOMO outputs an image per class where each pixel in the image is the centroid of the trained
+# object. So, we will get those output images and then run find_blobs() on them to extract the
+# centroids. We will also run get_stats() on the detected blobs to determine their score.
+# The Non-Max-Supression (NMS) object then filters out overlapping detections and maps their
+# position in the output image back to the original input image. The callback then returns a
+# list per class which each contain a list of (rect, score) tuples representing the detected
+# objects.
+
+
+def fomo_callback(model, rect):
+    out = model.output[0]
+    oh, ow, oc = model.output_shape
+    nms = tf.NMS(ow, oh, rect)
+    for i in range(oc):
+        img = out.get_image(i)
+        blobs = img.find_blobs(threshold_list, x_stride=1, area_threshold=1, pixels_threshold=1)
+        for b in blobs:
+            rect = b.rect()
+            x, y, w, h = rect
+            score = img.get_statistics(thresholds=threshold_list, roi=rect).l_mean() / 255.0
+            nms.add_bounding_box(x, y, x + w, y + h, score, i)
+    return nms.get_bounding_boxes()
```

Comment:
Should consider moving this post processing function to an extension module. The same for yolo, if we ever add an example.

Reply:
Yeah, I think edge impulse will need to generate that since FOMO is there model. This will serve as a template code for them to provide.

For yolo and other models... for the next gen system push we're probably going to want to pick a library of models we can leverage and then create an extension module with all the post processing callbacks.